### PR TITLE
Add support 2-byte arrays for utf with bom

### PR DESCRIPTION
**Pull request by rstm-sf**
Hello!

This pr support 2-byte arrays for utf with BOM.

https://github.com/CharsetDetector/UTF-unknown/blob/cc2b081a621001aedfac20229bd555585d366e9f/src/CharsetDetector.cs#L377

### CODE REVIEW

**Comment by rstm-sf** [comment]
Sorry, but Yoda notation poorly read

**Comment by rstm-sf** [comment]
@304NotModified, hello!

 If anything, then I can remove all the changes and leave only those similar to the following:
- `if (len < 2 || buf.Length < 2)`
- `if (len > 2 || buf.Length > 2)`

File: src/CharsetDetector.cs [hunk 1]
```diff
@@ -371,43 +371,39 @@ private void FindInputState(byte[] buf, int len)
...
+            if (len < 2 || buf.Length < 2)
+                return null;
+
+            var buf0 = buf[0];
+            var buf1 = buf[1];
+            
+            if (buf0 == 0xEF && buf1 == 0xBB && buf.Length > 2 && len > 2 && buf[2] == 0xBF)
+                return CodepageName.UTF8;
+
+            if (buf0 == 0xFE && buf1 == 0xFF)
```
Comment (304NotModified) [inline src/CharsetDetector.cs]: I think it's easier to read/maintain when nesting the ifs.

e.g.

`if (len > 3 && buf.Length >3){
...
   if( ....) 
   if( ....)
}
```

File: src/CharsetDetector.cs [hunk 2]
```diff
@@ -371,43 +371,39 @@ private void FindInputState(byte[] buf, int len)
...
-                            bomSet = CodepageName.X_ISO_10646_UCS_4_2143;
-                        break;
-                    case 0xFF:
-                        if (0xFE == b2 && 0x00 == b3 && 0x00 == b4)
-                            bomSet = CodepageName.UTF32_LE;
-                        else if (0xFE == b2)
-                            bomSet = CodepageName.UTF16_LE;
-                        break;
-                } // switch
+                // FE FF 00 00  UCS-4, unusual octet order BOM (3412)
```
Comment (304NotModified) [inline src/CharsetDetector.cs]: I really prefer some unit tests for these. If needed, you could make this method `internal`

Comment (rstm-sf) [reply src/CharsetDetector.cs]: They already exist ;)

<details>

https://github.com/CharsetDetector/UTF-unknown/blob/8b33ce8fe9de18ea4d02cef59a5455c0dfcc50ef/tests/CharsetDetectorTest.cs#L77-L80

https://github.com/CharsetDetector/UTF-unknown/blob/8b33ce8fe9de18ea4d02cef59a5455c0dfcc50ef/tests/CharsetDetectorTest.cs#L86-L89

https://github.com/CharsetDetector/UTF-unknown/blob/8b33ce8fe9de18ea4d02cef59a5455c0dfcc50ef/tests/CharsetDetectorTest.cs#L96-L100

https://github.com/CharsetDetector/UTF-unknown/blob/8b33ce8fe9de18ea4d02cef59a5455c0dfcc50ef/tests/CharsetDetectorTest.cs#L107-L111

https://github.com/CharsetDetector/UTF-unknown/blob/8b33ce8fe9de18ea4d02cef59a5455c0dfcc50ef/tests/CharsetDetectorTest.cs#L118-L121

https://github.com/CharsetDetector/UTF-unknown/blob/8b33ce8fe9de18ea4d02cef59a5455c0dfcc50ef/tests/CharsetDetectorTest.cs#L127-L130

https://github.com/CharsetDetector/UTF-unknown/blob/8b33ce8fe9de18ea4d02cef59a5455c0dfcc50ef/tests/CharsetDetectorTest.cs#L136-L139

</details>

Comment (304NotModified) [reply src/CharsetDetector.cs]: Why do we need this code then if no tests has changed? ;)

Comment (304NotModified) [reply src/CharsetDetector.cs]: This isn't a refactor, isn't?

File: src/CharsetDetector.cs (hunk 1 above)
Comment (rstm-sf) [reply src/CharsetDetector.cs]: Did I understand correctly?

File: src/CharsetDetector.cs (hunk 2 above)
Comment (rstm-sf) [reply src/CharsetDetector.cs]: Ahh. Wait a second!

Comment (rstm-sf) [reply src/CharsetDetector.cs]: Added tests

**Pull request merged**